test/db/SQLJSPackageDB.test.ts

```diff
@@ -16,19 +15,19 @@ describe('SQLJSPackageDB', () => {
   beforeEach(() => {
     loggerSpy.reset();
     sqlDb = new SQL.Database();
-    dbRunSpy = jest.spyOn(sqlDb, 'run');
```

Comment:
Now that the database is initialized in the class, we _can't_ spy on it because it's a private variable, so we don't have access to it from the test. If we wanted to spy on it, we'd need to either make the `database` member _public_ or we'd need to expose a constructor that takes a sql.js Database as an argument. I didn't like either of those options and decided that the spy was not strictly necessary for these tests since we are still able to validate the behavior in other ways. In addition, we weren't even checking the arguments to the spy anyway -- we were only checking that _some statement_ was executed (but not checking it was the right statement); so that seemed of limited value anyway.

That said, if you think it would be better to expose the member variable for testing purposes, I'm open to discussing that. I don't think we should add a constructor to take in `Database` though since that's exactly what we were trying to avoid in this PR.

Reply:
Ah, no that makes total sense! I think if maybe I had tried to use that spy I would have figured that out, so my bad! I don't think it's worth changing it just for this one check in the tests, so I'm good with it as is.